pkg/controllers/jobset_controller.go

```diff
@@ -412,14 +412,27 @@ func (r *JobSetReconciler) resumeJobsIfNecessary(ctx context.Context, js *jobset
 	}
 
 	startupPolicy := js.Spec.StartupPolicy
+
+	// Map where key is the Job name and value is the Job replicas.
+	rJobsReplicas := map[string]int32{}
+
 	// If JobSpec is unsuspended, ensure all active child Jobs are also
 	// unsuspended and update the suspend condition to true.
 	for _, replicatedJob := range js.Spec.ReplicatedJobs {
 		replicatedJobStatus := findReplicatedJobStatus(replicatedJobStatuses, replicatedJob.Name)
+
+		rJobsReplicas[replicatedJob.Name] = replicatedJob.Replicas
+
+		// For depends on, the Job is created only after the previous replicatedJob reached the status.
+		if replicatedJob.DependsOn != nil && !isDependsOnJobReachedStatus(replicatedJob.DependsOn[0], rJobsReplicas[replicatedJob.DependsOn[0].Name], replicatedJobStatuses) {
```

Comment:
I think this is a bug. You probably want the idx - 1 rather than 0.

Reply:
@andreyvelich there are some comments in this PR confused about the hardcoded index 0, it surprised me as well at first until I remembered the design discussion about single vs multi-dependency and future proofing. 

I think hard coding the index of a specific dependency makes the code a bit surprising/confusing as compared to simply looping through the slice checking all dependencies, and using the CEL validation you've added to enforce the single dependency only.

Reply:
I see, @danielvegamyhre do you suggest that in **this PR** we implement the `dependencyReachedStatus` function in a way that it can accept array and verify that all dependent ReplicatedJobs reach status ?